app/src/forms/form/exportService.js

```diff
@@ -133,7 +133,29 @@ const service = {
     };
   },
 
+  _readLatestFormSchema: async (formId) => {
+    return FormVersion.query()
+      .select('schema')
+      .withGraphFetched('submissions')
+      .where('formId', formId)
+      .modify('orderVersionDescending')
+      .first().then((row) => row.schema);
+  },
+
   _formatSubmissionsCsv: async (form, data) => {
+
+    // -- get column order to match field order in form design
+    // object key order is not preserved when submission JSON is saved to jsonb field type in postgres.
+    // get correctly ordered field names (keys) from latest form version
+    let latestFormDesign = await service._readLatestFormSchema(form.id);
+    let fieldNames = latestFormDesign.components
+      .filter(x => x.input === true)
+      .map(({key}) => key);
+
+    // get meta properties in 'form.<child.key>' string format
+    let metaKeys = Object.keys(data[0].form);
+    let metaHeaders = metaKeys.map(x => 'form.' + x);
```

Comment:
All 4 of these variables should be `const` as they are never reassigned to later in the block scope.